src/Datadog.Trace/Configuration/CsvMapConfigurationSource.cs

```diff
@@ -0,0 +1,60 @@
+using System;
+using System.Collections.Concurrent;
+
+namespace Datadog.Trace.Configuration
+{
+    /// <summary>
+    /// Represents a configuration source that retrieves values from
+    /// a comma-separate list of key-value pairs.
+    /// </summary>
+    public class CsvMapConfigurationSource : StringConfigurationSource
```

Comment:
Seems like we can drop the `StringConfigurationSource` base class since we are not using anything from it. Its main entry point (`GetString()`) even throws a `NotImplementedException`. If we do that, we can simplify this even further by making this a static helper class with a single method:
```csharp
ConcurrentDictionary<string, string> CsvMap.ParseConfig(string);
```

Reply:
Or, different approach, what about moving `ParseConfig()` to `StringConfigurationSource` and calling it `GetDictionary()`? The in the `TracerSettings` constructor we can do
```csharp
GlobalTags = source?.GetDictionary(ConfigurationKeys.GlobalTags);
```
which is more consistent with the other settings.

Reply:
I was writing a comment here and realized it was getting long enough to deserve it's own wiki page, so I enabled the Wiki feature in this repo and added its first page: [Tracer Configuration](https://github.com/DataDog/dd-trace-dotnet/wiki/Tracer-Configuration). I think this is a good way to document the code going forwards.

(Note that this is developer-centric documentation. [User documentation](https://docs.datadoghq.com/tracing/setup/dotnet/) will remain where it is for the time being.)

Reply:
👍

Reply:
@lucaspimentel your suggestion for moving the `ParseConfig` method to `StringConfigurationSource.GetDictionary` is a good one, but in order to have it seamlessly available via `source?.GetDictionary`, it would also have to be in `IConfigurationSource` which means it should probably be implemented in two other classes (Environment and Composite configuration source). I'm not that crazy about the latter part so I'll keep it just in `StringConfigurationSource` and cast. Let me know what you think.

Reply:
After attempting to work through adding a `GetDictionary` to `IConfigurationSource`, I've concluded that that doesn't seem like a workable approach. If we want to try to add this as a generalized configuration method, I think it will take a lot more thought.